tests/CribblyBackend.Core.UnitTests/Players/Services/PlayerServiceTests.cs

```diff
@@ -0,0 +1,153 @@
+            );
+            var testP2 = await _fakePlayerRepository.CreateAsync(
+                new Player { AuthProviderId = TestData.NewString(), Email = email2, Name = TestData.NewString() }
+            );
+
+            var p1 = await _playerService.GetByEmailAsync(email1);
+            Assert.Equal(1, p1.Id);
+            Assert.Equal(testP1.AuthProviderId, p1.AuthProviderId);
+            Assert.Equal(testP1.Email, p1.Email);
+            Assert.Equal(testP1.Name, p1.Name);
+            var p2 = await _playerService.GetByEmailAsync(email2);
+            Assert.Equal(2, p2.Id);
+            Assert.Equal(testP2.AuthProviderId, p2.AuthProviderId);
+            Assert.Equal(testP2.Email, p2.Email);
+            Assert.Equal(testP2.Name, p2.Name);
+        }
+
+        [Fact]
+        public async Task GetByAuthProviderIdShouldGetCorrectPlayer()
+        {
+            var authId1 = TestData.NewString();
+            var authId2 = TestData.NewString();
+            var testP1 = await _fakePlayerRepository.CreateAsync(
+                new Player { AuthProviderId = authId1, Email = TestData.NewString(), Name = "1" }
+            );
+            var testP2 = await _fakePlayerRepository.CreateAsync(
+                new Player { AuthProviderId = authId2, Email = TestData.NewString(), Name = "2" }
+            );
+
+            var p1 = await _playerService.GetByAuthProviderIdAsync(authId1);
+            Assert.Equal(1, p1.Id);
+            Assert.Equal(testP1.AuthProviderId, p1.AuthProviderId);
+            Assert.Equal(testP1.Email, p1.Email);
+            Assert.Equal(testP1.Name, p1.Name);
+            var p2 = await _playerService.GetByAuthProviderIdAsync(authId2);
+            Assert.Equal(2, p2.Id);
+            Assert.Equal(testP2.AuthProviderId, p2.AuthProviderId);
+            Assert.Equal(testP2.Email, p2.Email);
+            Assert.Equal(testP2.Name, p2.Name);
+        }
+
+        [Fact]
+        public async Task GetByAuthProviderIdShouldCacheResult()
+        {
+            var authId = TestData.NewString();
+            var testPlayer = await _fakePlayerRepository.CreateAsync(
+                new Player { AuthProviderId = authId, Email = TestData.NewString(), Name = TestData.NewString() }
+            );
+
+            var p = await _playerService.GetByAuthProviderIdAsync(authId);
+            Assert.Equal(authId, p.AuthProviderId);
+            Assert.Equal(testPlayer.Email, p.Email);
+            Assert.Equal(testPlayer.Name, p.Name);
+
+            _fakePlayerRepository.Update(new Player { Id = testPlayer.Id, AuthProviderId = authId, Name = "super not random", Email = "not random" });
+            var updatedPlayer = await _playerService.GetByIdAsync(testPlayer.Id);
+            Assert.Equal(authId, updatedPlayer.AuthProviderId);
+            Assert.Equal("not random", updatedPlayer.Email);
+            Assert.Equal("super not random", updatedPlayer.Name);
```

Comment:
Maybe a clearer way to test this would be for the `FakePlayerRepository` to be able to keep track of the number of calls for each method and just assert that the method to get a player only gets called once (because the player would be cached)